packages/dd-trace/src/exporters/agent/writer.js

```diff
@@ -1,129 +1,74 @@
 'use strict'
 
 const platform = require('../../platform')
-const log = require('../../log')
-const encode = require('../../encode')
 const tracerVersion = require('../../../lib/version')
+const msgpack = require('msgpack-lite')
 
-const MAX_SIZE = 8 * 1024 * 1024 // 8MB
-const METRIC_PREFIX = 'datadog.tracer.node.exporter.agent'
+const Writer05 = require('./writer-0.5')
+const Writer04 = require('./writer-0.4')
+const BaseWriter = require('./base-writer')
 
 class Writer {
   constructor (url, prioritySampler, lookup) {
-    this._url = url
-    this._prioritySampler = prioritySampler
-    this._lookup = lookup
+    this._appends = []
+    this._needsFlush = false
 
-    this._reset()
-  }
-
-  get length () {
-    return this._count
-  }
-
-  append (spans) {
-    log.debug(() => `Encoding trace: ${JSON.stringify(spans)}`)
-
-    this._encode(spans)
-  }
-
-  flush () {
-    if (this._count > 0) {
-      const data = platform.msgpack.prefix(this._buffer.slice(0, this._offset), this._count)
+    const fakeWriter = Object.create(BaseWriter.prototype)
```

Comment:
I don't understand the purpose of this.

Reply:
In hindsight, yeah this was not the best approach. The idea was to re-use BaseWriter's code as much as possible to build up the HTTP request. There are definitely better/cleaner ways to do this that make more intuitive sense. I'll work on it.